files/grest/rpc/stake_address/stake_distribution.sql

```diff
@@ -0,0 +1,156 @@
+CREATE TABLE IF NOT EXISTS GREST.STAKE_DISTRIBUTION (
+    STAKE_ADDRESS varchar PRIMARY KEY,
+    POOL_ID varchar,
+    TOTAL_BALANCE numeric,
+    UTXO numeric,
+    REWARDS numeric,
+    WITHDRAWALS numeric,
+    REWARDS_AVAILABLE numeric,
+    RESERVES numeric,
+    TREASURY numeric
+);
+
+CREATE TABLE IF NOT EXISTS GREST.CONTROL_TABLE (
+    control_type text PRIMARY KEY,
```

Comment:
Query type (eg: stake distribution), Block hash  (nullable in this case), Block ID

This control table can be filled by different queries with their properties, and easier to update row based on their last checks(?)

Reply:
This was totally the first draft only so I would expect changes here yeah. I remember us mentioning adding an entry for db-sync version for example. So I just made the entires as general as possible, i.e. control_type doesn't include just queries, and value is a text field so it can be filled with arbitrary data.

We can add block hash if we want to as a nullable field - unless we know only query entries will go in this table.

Reply:
I see - I guess the mapping done that way is easier to include bits indeed...in that case, we could add 3 columns:
key, last_value, artifacts - which could then include vvaried (JSON, content depending on query) things

Reply:
That sounds good, I'll update it here. Another thing I was thinking about we should have a centralised query where this control table is created.

We _can_ have it on every query that needs it, but then if we ever change the structure of it we would have to change it on every query instead of just one place.